### PR TITLE
feat(disk-balance):  init framework for adding disk balance automatically(1/3)

### PR DESCRIPTION
**Pull request by Shuo-Jia**
- **init framework for adding disk balance automatically**  https://github.com/pegasus-kv/admin-cli/pull/37
- implement framework for adding disk balance automatically https://github.com/pegasus-kv/admin-cli/pull/38
- complete framework for adding disk balance automatically https://github.com/pegasus-kv/admin-cli/pull/39

support balance target node disk usage automatically:
 1. change the pegasus server disk cleaner internal for clean temp replica to free disk space in time
 2. get the optimal migrate action to be ready to balance the disk until can't migrate base latest disk space stats
 3. if current replica is not `secondary` status, force assign the replica to `secondary` status
 4. migrate the replica base `getNextMigrateAction` result
 5. loop query migrate progress using `DiskMigrate`, it will response `ERR_BUSY` if running
 6. start next loop until can't allow to balance the node
 7. recover disk cleaner internal if balance complete
 8. set meta status to `lively` to balance primary and secondary

### CODE REVIEW

File: executor/disk_balance.go [hunk 1]
```diff
@@ -60,8 +61,105 @@ func DiskMigrate(client *Client, replicaServer string, pidStr string, from strin
...
+// -6. start next loop until can't allow to balance the node
+// -7. recover disk cleaner internal if balance complete
+// -8. set meta status to `lively` to balance primary and secondary // TODO(jiashuo1)
+func DiskBalance(client *Client, replicaServer string, minSize int64, auto bool) error {
+	err := changeDiskCleanerInterval(client, replicaServer, 1)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = changeDiskCleanerInterval(client, replicaServer, 86400)
```
Comment (Smityz) [inline executor/disk_balance.go:79]: 86400 is a magic number, maybe you should use a const num instead

File: executor/disk_balance.go [hunk 2]
```diff
@@ -60,8 +61,105 @@ func DiskMigrate(client *Client, replicaServer string, pidStr string, from strin
...
+			continue
+		}
+		err = DiskMigrate(client, replicaServer, action.replica.Gpid, action.from, action.to)
+		if err == nil {
+			fmt.Printf("migrate(%s) has started, wait complete...\n", action.toString())
+			for {
+				// TODO(jiashuo1): using DiskMigrate RPC to query status, consider support queryDiskMigrateStatus RPC
+				err = DiskMigrate(client, replicaServer, action.replica.Gpid, action.from, action.to)
+				if err == nil {
+					time.Sleep(time.Second * 10)
```
Comment (Smityz) [inline executor/disk_balance.go:105]: Why always sleep?
Maybe we can improve it by responding to the ball

File: executor/disk_balance.go (hunk 1 above)
Comment (Shuo-Jia) [reply executor/disk_balance.go:79]: why is a magic? the server default value is 84600

File: executor/disk_balance.go [hunk 3]
```diff
@@ -60,8 +61,105 @@ func DiskMigrate(client *Client, replicaServer string, pidStr string, from strin
...
+	}()
+
+	for {
+		action, err := getNextMigrateAction(client, replicaServer, minSize)
+		if err != nil {
+			return err
+		}
+		if action.replica.Status != "secondary" {
+			err := forceAssignReplicaToSecondary(client, replicaServer, action.replica.Gpid)
+			if err != nil {
```
Comment (Smityz) [inline executor/disk_balance.go:92]: this line can be combined with the above line

File: executor/disk_balance.go (hunk 2 above)
Comment (Shuo-Jia) [reply executor/disk_balance.go:105]: sleep for waiting loop, we can't  poll wiht high frequency

File: executor/disk_balance.go [hunk 4]
```diff
@@ -60,8 +61,105 @@ func DiskMigrate(client *Client, replicaServer string, pidStr string, from strin
...
+			fmt.Printf("migrate(%s) has started, wait complete...\n", action.toString())
+			for {
+				// TODO(jiashuo1): using DiskMigrate RPC to query status, consider support queryDiskMigrateStatus RPC
+				err = DiskMigrate(client, replicaServer, action.replica.Gpid, action.from, action.to)
+				if err == nil {
+					time.Sleep(time.Second * 10)
+					continue
+				}
+
+				if strings.Contains(err.Error(), "ERR_BUSY") {
```
Comment (Smityz) [inline executor/disk_balance.go:109]: "ERR_BUSY" -> base.ERR_BUSY.String()

File: executor/disk_balance.go (hunk 3 above)
Comment (levy5307) [reply executor/disk_balance.go:92]: +1

File: executor/disk_info.go [hunk 5]
```diff
@@ -31,22 +31,29 @@ import (
...
-	ReplicaCount DiskInfoType = 1
+	CapacitySize DiskInfoType = "CapacitySize"
+	ReplicaCount DiskInfoType = "ReplicaCount"
 )
 
 // QueryDiskInfo command
-// TODO(jiashuo1) need refactor
 func QueryDiskInfo(client *Client, infoType DiskInfoType, replicaServer string, tableName string, diskTag string) error {
+	_, err := queryDiskInfo(client, infoType, replicaServer, tableName, diskTag, true)
+	if err != nil {
```
Comment (levy5307) [inline executor/disk_info.go:44]: `return err` directly

**Review by levy5307 — APPROVED** [review]

**Comment by Smityz** [comment]
LGTM

**Pull request merged**